python/cudf/cudf/core/series.py

```diff
@@ -3933,238 +3933,6 @@ def count(self, level=None, **kwargs):
-
-        Examples
-        --------
-        >>> import cudf
-        >>> ser = cudf.Series([1, 5, 2, 4, 3])
-        >>> ser.cummin()
-        0    1
-        1    1
-        2    1
-        3    1
-        4    1
-        """
-
-        if axis not in (None, 0):
-            raise NotImplementedError("axis parameter is not implemented yet")
-
-        skipna = True if skipna is None else skipna
-
-        if skipna:
-            result_col = self.nans_to_nulls()._column
-        else:
-            result_col = self._column.copy()
-            if result_col.has_nulls:
-                # Workaround as find_first_value doesn't seem to work
-                # incase of bools.
-                first_index = int(
-                    result_col.isnull().astype("int8").find_first_value(1)
-                )
-                result_col[first_index:] = None
-
-        return Series(
-            result_col._apply_scan_op("min"), name=self.name, index=self.index,
-        )
-
-    def cummax(self, axis=0, skipna=True, *args, **kwargs):
-        """
-        Return cumulative maximum of the Series.
-
-        Parameters
-        ----------
-
-        skipna : bool, default True
-            Exclude NA/null values. If an entire row/column is NA,
-            the result will be NA.
-
-        Returns
-        -------
-        Series
-
-        Notes
-        -----
-        Parameters currently not supported is `axis`
-
-        Examples
-        --------
-        >>> import cudf
-        >>> ser = cudf.Series([1, 5, 2, 4, 3])
-        >>> ser.cummax()
-        0    1
-        1    5
-        2    5
-        3    5
-        4    5
-        """
-        if axis not in (None, 0):
-            raise NotImplementedError("axis parameter is not implemented yet")
-
-        skipna = True if skipna is None else skipna
-
-        if skipna:
-            result_col = self.nans_to_nulls()._column
-        else:
-            result_col = self._column.copy()
-            if result_col.has_nulls:
-                first_index = int(
-                    result_col.isnull().astype("int8").find_first_value(1)
-                )
-                result_col[first_index:] = None
-
-        return Series(
-            result_col._apply_scan_op("max"), name=self.name, index=self.index,
-        )
-
-    def cumsum(self, axis=0, skipna=True, *args, **kwargs):
-        """
-        Return cumulative sum of the Series.
-
-        Parameters
-        ----------
-
-        skipna : bool, default True
-            Exclude NA/null values. If an entire row/column is NA,
-            the result will be NA.
-
-
-        Returns
-        -------
-        Series
-
-        Notes
-        -----
-        Parameters currently not supported is `axis`
-
-        Examples
-        --------
-        >>> import cudf
-        >>> ser = cudf.Series([1, 5, 2, 4, 3])
-        >>> ser.cumsum()
-        0    1
-        1    6
-        2    8
-        3    12
-        4    15
```

Comment:
Can we preserve the docstring content & examples for `Series`/`Index` APIs that show up in docs? With all the implementations now relying in the `Frame` class, all the dedicated `Series` & `Index` page docstrings are going to show `DataFrame`/`Frame` docstrings.

Reply:
@galipremsagar thanks for bringing this up. I had been postponing dealing with this question through my refactorings, but we may as well address it now since it applies to all APIs that are shared by DataFrame and Series (and now Index as well). `pandas` docstrings for methods like these use phrasing like "returns Series or DataFrame" and they contain examples of operations on both types in both pages. Do we want to do the same for everything going forward? For example, here are the docs for [`pandas.Series.cumsum`](https://pandas.pydata.org/docs/reference/api/pandas.Series.cumsum.html) and [`pandas.DataFrame.cumsum`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.cumsum.html).

Reply:
I've mimicked pandas approach here by including both sets of examples in the Frame docstring.

Reply:
Sounds good to me, as a follow-up low priority thing we may have to explore templated docstrings like pandas does for our object types but again it can be a future TODO.

Reply:
Yes I agree, I have been thinking about this as well. Opened #9050 to discuss this.